starknet_py/net/full_node_client.py

```diff
@@ -101,6 +104,83 @@ async def get_block_traces(
+        follow_continuation_token: bool = False,
+        continuation_token: Optional[str] = None,
+        chunk_size: int = 1,
+    ) -> EventsResponse:
+        # pylint: disable=too-many-arguments
+        """
+        :param address: The address of the contract that emitted the event.
+            Must be in a form of RPC accepted felt.
+        :param keys: List of names of events that are searched for.
+            Must be in a form of RPC accepted felt after being hashed by `keccak` hash.
+        :param from_block_number: Number of the block from which events searched for **starts**
+            or literals `"pending"` or `"latest"`. Mutually exclusive with ``from_block_hash`` parameter.
+        :param from_block_hash: Hash of the block from which events searched for **starts**
+            or literals `"pending"` or `"latest"`. Mutually exclusive with ``from_block_number`` parameter.
+        :param to_block_number: Number of the block to which events searched for **end**
+            or literals `"pending"` or `"latest"`. Mutually exclusive with ``to_block_hash`` parameter.
+        :param to_block_hash: Hash of the block to which events searched for **end**
+            or literals `"pending"` or `"latest"`. Mutually exclusive with ``to_block_number`` parameter.
+        :param follow_continuation_token: Flag deciding whether all events should be collected during one function call,
+            defaults to False.
+        :param continuation_token: Continuation token from which the returned events start.
+        :param chunk_size: Size of chunk of events returned by one ``get_events`` call, defaults to 1 (minimum).
+
+        :return: ``EventsResponse`` dataclass containing events and optional continuation token.
+        """
+
+        if chunk_size <= 0:
+            raise ValueError("Argument chunk_size must be greater than 0.")
+        params = {
+            "chunk_size": chunk_size,
+            "from_block": get_small_block_identifier(
+                from_block_hash, from_block_number
+            ),
+            "to_block": get_small_block_identifier(to_block_hash, to_block_number),
+            "address": _to_rpc_felt(address),
+            "keys": keys,
+        }
+
+        if continuation_token is not None:
+            params["continuation_token"] = continuation_token
+        events_list = []
+        while True:
+            events, continuation_token = await self._get_events_chunk(params)
+            events_list.extend(events)
+            if not follow_continuation_token or continuation_token is None:
+                break
+            params["continuation_token"] = continuation_token
+        events_response = cast(
+            EventsResponse,
+            EventsSchema().load(
+                {"events": events_list, "continuation_token": continuation_token}
+            ),
+        )
+
+        return events_response
+
+    async def _get_events_chunk(
+        self,
+        params: dict,
```

Comment:
It can be changed to:
```python
    async def get_events(
        self,
        address: Hash,
        keys: List[str],
        *,
        from_block_number: Optional[Union[int, Tag]] = None,
        from_block_hash: Optional[Union[Hash, Tag]] = None,
        to_block_number: Optional[Union[int, Tag]] = None,
        to_block_hash: Optional[Union[Hash, Tag]] = None,
        follow_continuation_token: bool = False,
        continuation_token: Optional[str] = None,
        chunk_size: int = 1,
    ) -> EventsResponse:
        # pylint: disable=too-many-arguments
        """
        :param address: The address of the contract that emitted the event.
            Must be in a form of RPC accepted felt.
        :param keys: List of names of events that are searched for.
            Must be in a form of RPC accepted felt after being hashed by `keccak` hash.
        :param from_block_number: Number of the block from which events searched for **starts**
            or literals `"pending"` or `"latest"`. Mutually exclusive with ``from_block_hash`` parameter.
        :param from_block_hash: Hash of the block from which events searched for **starts**
            or literals `"pending"` or `"latest"`. Mutually exclusive with ``from_block_number`` parameter.
        :param to_block_number: Number of the block to which events searched for **end**
            or literals `"pending"` or `"latest"`. Mutually exclusive with ``to_block_hash`` parameter.
        :param to_block_hash: Hash of the block to which events searched for **end**
            or literals `"pending"` or `"latest"`. Mutually exclusive with ``to_block_number`` parameter.
        :param follow_continuation_token: Flag deciding whether all events should be collected during one function call,
            defaults to False.
        :param continuation_token: Continuation token from which the returned events start.
        :param chunk_size: Size of chunk of events returned by one ``get_events`` call, defaults to 1 (minimum).

        :return: ``EventsResponse`` dataclass containing events and optional continuation token.
        """

        if chunk_size <= 0:
            raise ValueError("Argument chunk_size must be greater than 0.")

        from_block = _get_raw_block_identifier(from_block_hash, from_block_number)
        to_block = _get_raw_block_identifier(to_block_hash, to_block_number)
        address = _to_rpc_felt(address)

        events_list = []
        while True:
            events, continuation_token = await self._get_events_chunk(from_block, to_block, address, keys, chunk_size, continuation_token)
            events_list.extend(events)
            if not follow_continuation_token or continuation_token is None:
                break

        events_response = cast(
            EventsResponse,
            EventsSchema().load(
                {"events": events_list, "continuation_token": continuation_token}
            ),
        )

        return events_response

    async def _get_events_chunk(
        self,
        from_block: Union[dict, Hash, Tag, None],
        to_block: Union[dict, Hash, Tag, None],
        address: Hash,
        keys: List[str],
        chunk_size: int,
        continuation_token: Optional[str] = None,
    ) -> Tuple[list, Optional[str]]:
        params = {
            "chunk_size": chunk_size,
            "from_block": from_block,
            "to_block": to_block,
            "address": address,
            "keys": keys,
        }
        if continuation_token is not None:
            params["continuation_token"] = continuation_token

        res = await self._client.call(
            method_name="getEvents",
            params={"filter": params},
        )

        if "continuation_token" in res:
            return res["events"], res["continuation_token"]
        return res["events"], None
```
What do you think? cc @war-in @drknzz